js/ui/overlay/overlay_position_controller.js

```diff
@@ -0,0 +1,492 @@
+
+
+        this.restorePositionOnNextRender(shouldRestorePosition);
+    }
+
+    resizeHandled() {
+        const shouldRestorePosition = this._props.restorePosition.onDimensionChangeAfterResize
+            || this._props.restorePosition.always;
+
+
+        this.restorePositionOnNextRender(shouldRestorePosition);
+    }
+
+    updateTarget(target) {
+        this._props.target = target;
+
+        this.updatePosition(this._props.position, target);
+    }
+
+    updatePosition(positionProp, targetProp = this._props.target) {
+        this._props.position = positionProp;
+        this._position = this._normalizePosition(positionProp, targetProp);
+
+        this._updateWrapperCoveredElement();
+    }
+
+    updateContainer(containerProp) {
+        this._props.container = containerProp;
+
+        const container = containerProp ?? viewPort();
+
+        let $container = this._$root.closest(container);
+
+        if(!$container.length) {
+            $container = $(container).first();
+        }
+
+        this._$container = $container.length ? $container : this._$root.parent();
+
+        this._updateWrapperCoveredElement();
+        this._updateDragResizeContainer();
+    }
+
+    detectVisualPositionChange(event) {
+        this._updateVisualPositionValue();
+        this._raisePositionedEvents(event);
+    }
+
+    positionContent() {
+        if(this._shouldRenderContentInitialPosition) {
+            this._renderContentInitialPosition();
+        } else {
+            move(this._$content, this._visualPosition);
+            this.detectVisualPositionChange();
+        }
+    }
+
+    positionWrapper() {
+        if(this._$wrapperCoveredElement) {
+            positionUtils.setup(this._$wrapper, { my: 'top left', at: 'top left', of: this._$wrapperCoveredElement });
+        }
+    }
+
+    isAllWindowCoveredByWrapper() {
+        return !this._$wrapperCoveredElement || isWindow(this._$wrapperCoveredElement.get(0));
+    }
+
+    styleWrapperPosition() {
+        const useFixed = this.isAllWindowCoveredByWrapper() || this._props._fixWrapperPosition;
+
+        const positionStyle = useFixed ? 'fixed' : 'absolute';
+        this._$wrapper.css('position', positionStyle);
+    }
+
+    _updateVisualPositionValue() {
+        this._previousVisualPosition = this._visualPosition;
+        this._visualPosition = locate(this._$content);
+    }
+
+    _renderContentInitialPosition() {
+        this._renderBoundaryOffset();
+        resetPosition(this._$content);
+        const resultPosition = positionUtils.setup(this._$content, this._position);
+        this._initialPosition = resultPosition;
+        this.detectVisualPositionChange();
+    }
+
+    _raisePositionedEvents(event) {
+        const previousPosition = this._previousVisualPosition;
+        const newPosition = this._visualPosition;
+
+        const isVisualPositionChanged = previousPosition?.top !== newPosition.top
+            || previousPosition?.left !== newPosition.left;
```

Comment:
```suggestion
        const previousPosition = this._previousVisualPosition || {};
        const newPosition = this._visualPosition;

        const isVisualPositionChanged = previousPosition.top !== newPosition.top
            || previousPosition.left !== newPosition.left;
```

Reply:
I don't think it's a good idea, because we don't wanna see `previousPosition: {}` on first positioning